### PR TITLE
chore(staff): Let staff access user permission endpoints

### PR DESCRIPTION
**Pull request by schew2381**
Let staff access endpoints to manage user permissions

Requires https://github.com/getsentry/sentry/pull/64429

### CODE REVIEW

File: tests/sentry/api/endpoints/test_user_permission_details.py [hunk 1]
```diff
@@ -9,68 +13,149 @@ class UserDetailsTest(APITestCase):
 
     def setUp(self):
         super().setUp()
-        self.user = self.create_user(is_superuser=True)
```
Comment (schew2381) [inline tests/sentry/api/endpoints/test_user_permission_details.py:12]: Changed the name of the default user in the tests from `self.user` to `self.superuser`

File: tests/sentry/api/endpoints/test_user_permission_details.py [hunk 2]
```diff
@@ -9,68 +13,149 @@ class UserDetailsTest(APITestCase):
...
-        resp = self.get_response("me", "broadcasts.admin")
-        assert resp.status_code == 403
+    # For each request method testcase, ensure regular users fail
+    def test_fails_without_superuser_or_superuser(self):
+        self.login_as(self.normal_user)
+        response = self.get_response("me", "broadcasts.admin")
+        assert response.status_code == 403
 
+    # For each request method testcase, ensure superuser+staff without users.admin fail
     def test_fails_without_users_admin_permission(self):
```
Comment (schew2381) [inline tests/sentry/api/endpoints/test_user_permission_details.py:31]: These two tests run for every request method testcase

File: tests/sentry/api/endpoints/test_user_permission_details.py [hunk 3]
```diff
@@ -9,68 +13,149 @@ class UserDetailsTest(APITestCase):
...
-        assert resp.status_code == 404
+        self.get_error_response("me", "broadcasts.admin", status_code=404)
+        # ensure we fail the scope check and call is_active_staff
+        assert mock_has_permission.call_count == 1
 
 
 class UserPermissionDetailsPostTest(UserDetailsTest):
     method = "POST"
 
-    def test_with_permission(self):
```
Comment (schew2381) [inline tests/sentry/api/endpoints/test_user_permission_details.py:48]: This was testing creating a duplicate permission, so renamed to `test_superuser_duplicate_permission`

File: tests/sentry/api/endpoints/test_user_permission_details.py [hunk 4]
```diff
@@ -9,68 +13,149 @@ class UserDetailsTest(APITestCase):
...
-        assert resp.status_code == 403
+        self.normal_user = self.create_user(is_superuser=False, is_staff=False)
 
-        self.user.update(is_superuser=True)
-        resp = self.get_response("me", "broadcasts.admin")
-        assert resp.status_code == 403
+    # For each request method testcase, ensure regular users fail
+    def test_fails_without_superuser_or_staff(self):
+        self.login_as(self.normal_user)
+        response = self.get_response("me", "broadcasts.admin")
```
Comment (cathteng) [inline tests/sentry/api/endpoints/test_user_permission_details.py:27]: nit: use `get_error_response`?

**Review by cathteng — APPROVED** [review]

**Pull request merged**